lib/media/track/localaudiotrack.js

```diff
@@ -20,14 +23,115 @@ const LocalMediaAudioTrack = mixinLocalMediaTrack(AudioTrack);
  * @emits LocalAudioTrack#started
  * @emits LocalAudioTrack#stopped
  */
+
 class LocalAudioTrack extends LocalMediaAudioTrack {
   /**
    * Construct a {@link LocalAudioTrack} from a MediaStreamTrack.
    * @param {MediaStreamTrack} mediaStreamTrack - An audio MediaStreamTrack
    * @param {LocalTrackOptions} [options] - {@link LocalTrack} options
    */
   constructor(mediaStreamTrack, options) {
+    // NOTE(mpatwardhan): by default workaround for WebKitBug1208516 will be enabled on Safari browsers
+    // although the bug is seen  mainly on iOS devices, we do not have a reliable way to tell iOS from MacOs
+    // userAgent on iOS pretends its macOs if Safari is set to request desktop pages.
+    const workaroundWebKitBug1208516 = guessBrowser() === 'safari';
+
+    options = Object.assign({
+      workaroundWebKitBug1208516,
+      getUserMedia
+    }, options);
+
     super(mediaStreamTrack, options);
+    Object.defineProperties(this, {
+      _watchCleanup: {
+        value: null,
+        writable: true
+      },
+      _getUserMedia: {
+        value: options.getUserMedia,
+      },
+      _didCallStop: {
+        value: false,
+        writable: true
+      }
+    });
+
+    // NOTE(mpatwardhan): as a workaround for webkitbug208516(https://bugs.webkit.org/show_bug.cgi?id=208516)
+    // start watching document visibility and reacquire audio track if ended or muted.
+    if (options.workaroundWebKitBug1208516) {
+      this._startWatching(mediaStreamTrack);
+    }
+  }
+
+  _shouldAttemptRestart() {
+    const isDocumentVisible = document && document.visibilityState === 'visible';
+    return isDocumentVisible && (this.mediaStreamTrack.muted || this.mediaStreamTrack.readyState === 'ended');
+  }
+
+  _acquireTrack() {
+    const mediaStreamPromise = this._getUserMedia({ audio: true, video: false });
```

Comment:
@makarandp0 We can't just use `{ audio: true }` here. We should make sure existing constraints are used:
```js
_getUserMedia({ audio: this.mediaStreamTrack.getConstraints() });
```